src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_VALUE = 10;
+    private static final double MORE_SIZE = 1.5;
+
+    private Object[] array;
+
+    private int size;
+
+    public ArrayList() {
+        array = new Object[DEFAULT_VALUE];
+        size = 0;
+    }
+
+    private void ensureCapacity() {
+        if (size == array.length) {
+            int extension = (int) (array.length * MORE_SIZE);
+            Object[] newArray = new Object[extension];
+            for (int i = 0; i < size; i++) {
+                newArray[i] = array[i];
+            }
+            array = newArray;
+        }
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index: " + index);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index != size) {
+            checkIndex(index);
+        }
+        ensureCapacity();
+        size++;
+        ensureCapacity();
 
+        for (int i = size - 1; i > index; i--) {
+            array[i] = array[i - 1];
+        }
```

Comment:
Please replace with predefined method according to checklist:
https://github.com/mate-academy/jv-array-list/blob/master/checklist.md#use-systemarraycopy-to-move-your-array-elements